extra_data/write_cxi.py

```diff
@@ -38,31 +40,65 @@ def __init__(self, detdata):
             frame_counts.index.values, frame_counts.values.astype(np.intp)
         )
 
-        # cumulative sum gives the end of each train, subtract to get start
+        # Cumulative sum gives the end of each train, subtract to get start
         self.train_id_to_ix = frame_counts.cumsum() - frame_counts
 
     @property
     def nmodules(self):
+        """
+        Number of detector modules.
+        """
```

Comment:
```suggestion
        """Number of detector modules."""
```

This is only a suggestion - I'm not going to get into battles over code style. But I think there's some value in using fewer lines where it doesn't hurt readability, because it means you can see more code on screen at once.